src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -1441,17 +1441,16 @@ void ClientChannel::RemoveResolverQueuedCall(ResolverQueuedCall* to_remove,
 void ClientChannel::UpdateServiceConfigInControlPlaneLocked(
     RefCountedPtr<ServiceConfig> service_config,
     RefCountedPtr<ConfigSelector> config_selector, const char* lb_policy_name) {
-  UniquePtr<char> service_config_json(
-      gpr_strdup(service_config->json_string().c_str()));
+  std::string service_config_json(service_config->json_string());
   if (GRPC_TRACE_FLAG_ENABLED(grpc_client_channel_routing_trace)) {
     gpr_log(GPR_INFO,
             "chand=%p: resolver returned updated service config: \"%s\"", this,
-            service_config_json.get());
+            service_config_json.c_str());
   }
   // Save service config.
   saved_service_config_ = std::move(service_config);
   // Swap out the data used by GetChannelInfo().
-  UniquePtr<char> lb_policy_name_owned(gpr_strdup(lb_policy_name));
+  std::string lb_policy_name_owned = lb_policy_name;
```

Comment:
How about just changing this method to take the `lb_policy_name` parameter as a `std::string`, so it doesn't need to convert internally?